core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java

```diff
@@ -139,17 +139,22 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
 
   @Override
   public void write(scala.collection.Iterator<Product2<K, V>> records) throws IOException {
-    boolean success = false;
     try {
       while (records.hasNext()) {
         insertRecordIntoSorter(records.next());
       }
       closeAndWriteOutput();
-      success = true;
-    } finally {
-      if (!success) {
-        sorter.cleanupAfterError();
+    } catch (Exception e) {
```

Comment:
AFAIK my original intent was to also ensure that we clean up after Throwables as well without necessarily having to catch throwable, so I think that's why I had the slightly unclear `success` boolean.